File: practicas/practica4_rb/lib/PowerEfficientSpaceStation.rb
#encoding: utf-8
require_relative 'SpaceStation'
require_relative 'PowerEfficientSpaceStationToUI'

module Deepspace
  class PowerEfficientSpaceStation < SpaceStation
    @@EFFICIENCYFACTOR = 1.10

    def initialize(station)
      fromOther(station)
    end

    def fire
      super*@@EFFICIENCYFACTOR
    end

    def protection
      super*@@EFFICIENCYFACTOR
    end

    def setLoot(l)
      res = super(l)
      if res == Transformation::SPACECITY
        res = Transformation::NOTRANSFORM
      end
    end

    def getUIversion
      PowerEfficientSpaceStationToUI.new self
    end

  end
end

File: practicas/practica4/src/main/java/deepspace/PowerEfficientSpaceStation.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author mario
 */
public class PowerEfficientSpaceStation extends SpaceStation{
    private final static float EFFICIENCYFACTOR = 1.1f;

    public PowerEfficientSpaceStation(SpaceStation station) {
        super(station);
    }
    
    @Override
    public Transformation setLoot(Loot loot) {
        Transformation t = super.setLoot(loot);
        if(t==Transformation.GETEFFICIENT){
            return t;
        }
        return Transformation.NOTRANSFORM;
    }
    
    @Override
    public float fire() {
        return super.fire() * EFFICIENCYFACTOR;
    }
    
    @Override
    public float protection() {
        return super.protection() * EFFICIENCYFACTOR;
    }
    
    @Override
    public SpaceStationToUI getUIversion() {
        return new PowerEfficientSpaceStationToUI(this);
    }
    
}
